tests/test_pmft.py

```diff
@@ -682,14 +683,76 @@ def test_orientation_with_query_points(self):
 
         # If we swap the order of the points, the angle should no longer
         # matter.
-        pmft.compute((box, points2), angles, points,
-                     neighbors={'mode': 'nearest', 'num_neighbors': 1})
         bond_vector = rowan.rotate(rowan.conjugate(quats), points2 - points)
         bins = ((bond_vector+max_width)*cells_per_unit_length/2).astype(int)
+
+        pmft.compute((box, points2), angles, points,
+                     neighbors={'mode': 'nearest', 'num_neighbors': 1})
         npt.assert_array_equal(
             np.asarray(np.where(pmft.bin_counts)).squeeze(),
             bins.squeeze()[:2])
 
+    def test_orientation_with_fewer_query_points(self):
+        """The orientations should be associated with the query points if they
+        are provided. Ensure that this works when the number of points and
+        query points differ."""
+        boxSize = 8
+        box = freud.box.Box.square(boxSize)
+        # Don't place the points at exactly distances of 0/1 apart to avoid any
+        # ambiguity when the distances fall on the bin boundaries.
+        points = np.array([[0.1, 0.1, 0], [0.89, 0.89, 0]],
+                          dtype=np.float32)
+        points2 = np.array([[1, 0, 0]],
+                           dtype=np.float32)
+        angles = np.array([np.deg2rad(0)]*points.shape[0], dtype=np.float32)
+        angles2 = np.array([np.deg2rad(0)]*points2.shape[0], dtype=np.float32)
+
+        def points_to_set(bin_counts):
+            """Extract set of unique occupied bins from pmft bin counts."""
+            return set(list(zip(*np.asarray(np.where(bin_counts)).tolist())))
```

Comment:
Is the `list(...)` necessary? I'd think `set(zip(...))` would be enough.
```suggestion
            return set(zip(*np.asarray(np.where(bin_counts)).tolist()))
```